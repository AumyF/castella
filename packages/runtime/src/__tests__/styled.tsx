import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";
import { styledComponent } from "..";

beforeEach(() => {
  document.head.innerHTML = "";
});

describe("styledComponent", () => {
  it("input", () => {
    const Hello = styledComponent({
      css: ".test-foobar{color:red;}",
      className: "test-foobar",
      elementName: "input",
    });

    render(<Hello />);

    expect(document.head.innerHTML).toMatchSnapshot();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it("div", () => {
    const Hello = styledComponent({
      css: ".test-divdiv{color:red;}",
      className: "test-divdiv",
      elementName: "div",
    });

    render(
      <Hello>
        <p>Hi!</p>
      </Hello>
    );

    expect(document.head.innerHTML).toMatchSnapshot();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
  it("rendered twice", () => {
    const Hello = styledComponent({
      css: ".test-divdiv{color:red;}",
      className: "test-divdiv",
      elementName: "div",
    });

    render(
      <Hello>
        <p>Hi!</p>
        <Hello>
          <p>Hoy!</p>
        </Hello>
      </Hello>
    );

    expect(document.head.innerHTML).toMatchSnapshot();
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
