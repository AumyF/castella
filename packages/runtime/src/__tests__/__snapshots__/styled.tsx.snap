// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styledComponent div 1`] = `"<style data-castella-styled=\\"test-divdiv\\">.test-divdiv{color:red;}</style>"`;

exports[`styledComponent div 2`] = `"<div><div class=\\"test-divdiv\\"><p>Hi!</p></div></div>"`;

exports[`styledComponent input 1`] = `"<style data-castella-styled=\\"test-foobar\\">.test-foobar{color:red;}</style>"`;

exports[`styledComponent input 2`] = `"<div><input class=\\"test-foobar\\"></div>"`;

exports[`styledComponent rendered twice 1`] = `"<style data-castella-styled=\\"test-divdiv\\">.test-divdiv{color:red;}</style>"`;

exports[`styledComponent rendered twice 2`] = `"<div><div class=\\"test-divdiv\\"><p>Hi!</p><div class=\\"test-divdiv\\"><p>Hoy!</p></div></div></div>"`;
